src/components/core/geoChart/GeoChartLegendRenderer.tsx

```diff
@@ -1,216 +1,84 @@
-import { stringToFloat } from "../../../helpers/utils";
 import PushpinSizeLegend from "./legends/PushpinSizeLegend";
 import { getGeoDataIndex } from "../../../helpers/geoChart";
 import { IGeoConfig, IGeoDataIndex } from "../../../interfaces/GeoChart";
-import { COLOR, SIZE, SEGMENT_BY } from "../../../constants/bucketNames";
+import { SIZE } from "../../../constants/bucketNames";
 import { TOP } from "../../visualizations/chart/legend/PositionTypes";
-import Legend, { ILegendProps } from "../../visualizations/chart/legend/Legend";
 
-export interface IChartLegendProps {
+export interface IGeoChartLegendRendererProps {
     config: IGeoConfig;
     execution: Execution.IExecutionResponses;
     locale: string;
     position?: string;
 }
-export default class GeoChartLegendRenderer extends React.PureComponent<IChartLegendProps> {
-    constructor(props: IChartLegendProps) {
-        super(props);
-    }
-
+export default class GeoChartLegendRenderer extends React.PureComponent<IGeoChartLegendRendererProps> {
+    public static defaultProps: Partial<IGeoChartLegendRendererProps> = {
+        config: {},
+    };
     public render() {
         const {
-            execution: { executionResult },
+            execution: { executionResult, executionResponse },
             config: { mdObject: { buckets = [] } = {} },
             locale,
             position = TOP,
         } = this.props;
 
-        const format: string = "#,##0.00"; // TODO: will implement with SegmentBy or Color Legend
         const geoDataIndex: IGeoDataIndex = getGeoDataIndex(buckets);
         const data = executionResult.data as Execution.DataValue[][];
         const classes = ["geo-legend", `position-${position}`];
         return (
             <div className={classNames(...classes)}>
-                {(geoDataIndex.hasOwnProperty(COLOR) || geoDataIndex.hasOwnProperty(SEGMENT_BY)) &&
-                    this.renderPushpinColorOrSegmentByLegend(geoDataIndex.hasOwnProperty(SEGMENT_BY))}
-                {geoDataIndex.hasOwnProperty(SIZE) && (
-                    <IntlWrapper locale={locale}>
-                        <IntlTranslationsProvider>
-                            {(props: ITranslationsComponentProps) =>
-                                this.renderPushpinSizeLegend(
-                                    (data[geoDataIndex.size] || []).map(stringToFloat),
-                                    format,
-                                    props.numericSymbols,
-                                )
-                            }
-                        </IntlTranslationsProvider>
-                    </IntlWrapper>
-                )}
+                {geoDataIndex.hasOwnProperty(SIZE) &&
+                    this.renderPushpinSizeLegend(
+                        (data[geoDataIndex.size] || []).map(stringToFloat),
+                        this.getFormatFromExecution(geoDataIndex.size, executionResponse),
+                        locale,
+                    )}
```

Comment:
it is not consistent between `geoDataIndex.hasOwnProperty(SIZE)` and `geoDataIndex.size`.
Please use 1 way only